### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Limits.Shapes.WideEqualizers

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/Shapes/WideEqualizers.lean [hunk 1]
```diff
@@ -0,0 +1,836 @@
...
+    (Trident.ofι ι w).ι = ι :=
+  rfl
+#align category_theory.limits.trident.ι_of_ι CategoryTheory.Limits.Trident.ι_ofι
+
+theorem Cotrident.π_ofπ [Nonempty J] {P : C} (π : Y ⟶ P) (w : ∀ j₁ j₂, f j₁ ≫ π = f j₂ ≫ π) :
+    (Cotrident.ofπ π w).π = π :=
+  rfl
+#align category_theory.limits.cotrident.π_of_π CategoryTheory.Limits.Cotrident.π_ofπ
+
+-- porting note: @[simp] as the linter complains even if we increase the priority
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Limits/Shapes/WideEqualizers.lean:278]: This wasn't a `simp` lemma in mathlib3, was it? I'm not sure why we have this porting note.

File: Mathlib/CategoryTheory/Limits/Shapes/WideEqualizers.lean [hunk 2]
```diff
@@ -0,0 +1,836 @@
...
+@[reassoc]
+theorem Trident.condition (j₁ j₂ : J) (t : Trident f) : t.ι ≫ f j₁ = t.ι ≫ f j₂ := by
+  rw [t.app_zero, t.app_zero]
+#align category_theory.limits.trident.condition CategoryTheory.Limits.Trident.condition
+
+@[reassoc (attr := simp)]
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Limits/Shapes/WideEqualizers.lean:284]:
```suggestion
@[reassoc]
```

**Comment by semorrison** [comment]
bors d+

File: Mathlib/CategoryTheory/Limits/Shapes/WideEqualizers.lean (hunk 1 above)
Comment (joelriou) [reply Mathlib/CategoryTheory/Limits/Shapes/WideEqualizers.lean]: It seems it had tagged an a `simp` lemma earlier in the port process. I have just fixed this.

Comment (mo271) [reply Mathlib/CategoryTheory/Limits/Shapes/WideEqualizers.lean]: LGTM!

**Pull request closed without merging**